_episodes/16-parallel.md

```diff
@@ -436,40 +345,39 @@ rank = comm.Get_rank()
 immediately before the `n_samples` line to set up the MPI environment for
 each process.
 
-We replace the `start_time` and `counts` lines with the lines:
+
+Immediately below these lines, let's distribute the work among the ranks
+with a calculation, that uses integer division and then distributes the 
+remaining work piece by piece
+* collect each rank's results into a `counts` variable on rank 0 using MPI
+  `reduce`.
+
+by adding the following three lines:
 
 ```
-if rank == 0:
-  start_time = datetime.datetime.now()
-  partitions = [ int(n_samples / cpus) ] * cpus
-  counts = [ int(0) ] * cpus
-else:
-  partitions = None
-  counts = None
+my_samples = n_samples//size
+if ( N % size > rank ):
```

Comment:
May want a small comment around the `//` and `%` lines, since those operators only show up in the Python gapminder lessons with sufficient explanation. In the inflammation lessons, `%` shows up once with very little explanation, and `//` is omitted entirely.

I'm also mentally working my way through the logic for lines 359-360 in general.

Reply:
Thanks.

Reply:
Thanks for the suggestion. Further explanation added.